base/sparse/cholmod.jl

```diff
@@ -87,22 +87,22 @@ function __init__()
             tmp = Array(Cint, 3)
             ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), version_array)
             ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), tmp)
-            if tmp != version_array
+            if tmp[1] != version_array[1]
```

Comment:
OK, I've made some tests, and I wasn't able to get the tests to pass with anything below SuiteSparse 4.2.0. With 4.1.0 I get this error (after rebuilding suitesparse-wrapper and the Julia image):
```
     * sparse               /home/milan/Dev/julia/usr/bin/julia: symbol lookup error: /home/milan/Dev/julia/usr/bin/../lib/libcholmod.so: undefined symbol: dpotrf_
```

This symbol is present in liblapack and can be resolved from Julia, so I'm not sure what's going on. If there's any chance this might work somewhere, I can try to get past it, else we can decide to stick to that. It includes libcholmod 2.1.1 (the second release with `cholmod_version()`), which means we would really be able to detect when the version is too old.

I've also discovered that the version check didn't even work below 4.1.0. With libcholmod < 2.1.0, i.e. when   `cholmod_version()` does not exist, `Libdl.dlsym` raised an error, while the code expected it to return `C_NULL`. That's a typical case of a rare code path that we supposedly care about but nobody tested. I've changed it to `try`/`catch`.

Does the logic in the new commit sound correct to you?